functional-samples/sample.co2meter/co2-state-iframe.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  CO2_METER_UNAVAILABLE,
+  CO2_METER_AVAILABLE,
+  NEW_CO2_READING,
+  NEW_TEMP_READING,
+  READING_UNKNOWN
+} from './modules/constant.js';
+import CO2Meter from './modules/co2_meter.js';
+
+window.onload = async () => {
+  // Register for messages to update chart upon new data readings.
+  chrome.runtime.connect().onMessage.addListener((msg) => {
+    if (msg.type === NEW_CO2_READING) {
```

Comment:
maybe: a switch expression might be easier to read.

Reply:
Done.